lib/libxdp/libxdp.c

```diff
@@ -2570,3 +2572,139 @@ bool xdp_multiprog__is_legacy(const struct xdp_multiprog *mp)
 {
 	return !!(mp && mp->is_legacy);
 }
+
+static void xdp_get_ifindex_prog_id(int ifindex, __u32 *prog_id)
+{
+	struct xdp_link_info xinfo = {};
+
+	if (!bpf_get_link_xdp_info(ifindex, &xinfo, sizeof(xinfo), 0)) {
+		switch (xinfo.attach_mode) {
+		case XDP_ATTACHED_SKB:
+			*prog_id = xinfo.skb_prog_id;
+			break;
+		case XDP_ATTACHED_DRV:
+			*prog_id = xinfo.drv_prog_id;
+			break;
+		case XDP_ATTACHED_MULTI:
+			*prog_id = xinfo.drv_prog_id ?: xinfo.skb_prog_id;
+			break;
+		case XDP_ATTACHED_NONE:
+		case XDP_ATTACHED_HW:
+		default:;
+		}
+	}
+}
+
+static int remove_pin_dir(const char *subdir)
+{
+	char prog_path[PATH_MAX], pin_path[PATH_MAX];
+	int err;
+	DIR *d = NULL;
```

Comment:
We generally try to use the 'reverse x-mas tree' variable ordering from the kernel netdev tradition